Player Controller script with shooting

Iam-B0B opened the pull request:
https://trello.com/c/IjB8xVNV/23-playercontroller

Player Controller Script 

The player shooting script is within the player controller 
- The player needs to have a projectile prefab 
- projectile launch point

The projectile script 
- Allows for the projectile to be destroyed when the enemy is missed
- Will not destroy enemy

Iam-B0B committed:
Player Controller script with shooting

Player Controller Script With player shooting

MV1090 reviewed (approved):
Overall looks good, from a personal note I'd change the variables in the player script to private and [SerializeField] them as they don't need to be accessed from anywhere else (unless I've missed something) , not necessary so I guess its up to you if you want to change it.

SolomanDavis commented on Assets/_Scripts/Controllers/PlayerController.cs:
```diff
@@ -0,0 +1,40 @@
+    public float moveSpeed = 5f; // Speed of player movement
+    public GameObject bulletPrefab; // Prefab of the bullet GameObject
+    public Transform firePoint; // Point where the bullet will be spawned
+    public float bulletSpeed = 10f; // Speed of the bullet
```
Agree with Mark, setting these to private means they aren't accessed by classes outside of this one and adding `[SerializeField]` means we can update them from the unity editor which will be useful when tweaking things to make them look right next week

Iam-B0B replied:
all have been set to [SerializeField] and most of the public has been changed to private

SolomanDavis commented on Assets/_Scripts/Controllers/PlayerController.cs:
```diff
@@ -0,0 +1,40 @@
+        // Instantiate a bullet at the firePoint position
+        GameObject bullet = Instantiate(bulletPrefab, firePoint.position, Quaternion.identity);
+
+        // Get the Rigidbody2D component of the bullet
+        Rigidbody2D rb = bullet.GetComponent<Rigidbody2D>();
+
+        // Apply velocity to the bullet (upward)
+        rb.velocity = Vector2.up * bulletSpeed;
```
If there is going to be a separate script for the Player Projectile, the rigidbody updates for it should be in the projectile script and not in the PlayerController. The PlayerController shouldn't know anything about how the bullet moves, only that it can shoot.

Iam-B0B replied:
has been moved to its script titled Shoot as the player controller no longer has control

SolomanDavis commented on Assets/_Scripts/Projectile/Projectile.cs:
```diff
@@ -0,0 +1,19 @@
+using UnityEngine;
+
+[RequireComponent(typeof(Rigidbody2D))]
+public class Projectile : MonoBehaviour
```
I noticed that the Enemy abstract class PR also contains a separate `EnemyProjectile` [here](https://github.com/SolomanDavis/VGP145-S24-capstone-retro/pull/24/files#diff-0c7a0f53c33fe53b2d7092c22fa921f95b6f4154191e0391e847bf4bf83d05a9).

I believe they both act the same as in they both move at the exact same speed and only vertically. There's only 2 differences:
- PlayerProjectile moves upwards while EnemyProjectile moves downwards.
- PlayerProjectile kills enemies while EnemyProjectile kills player.

Once we merge this in it might be worth looking into consolidating them into one projectile script and updating the logic appropriately. Worth considering when we get to testing and want to update values of both these projectiles and ensure they perform the same.

(can leave this for now but something to come back to)

Iam-B0B committed:
Merge branch 'master' into Kyle

Iam-B0B committed:
Changes to the player contoller

Added serialized feilds and private values have been assigned.

Iam-B0B committed:
Changes to Player Death

Player animations were chnaged to make similar to the heavy enemy death animation
